DfE.FindInformationAcademiesTrusts.Data.AcademiesDb/DataSourceProvider.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Diagnostics.CodeAnalysis;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+
+namespace DfE.FindInformationAcademiesTrusts.Data.AcademiesDb;
+
+public class DataSourceProvider : IDataSourceProvider
+{
+    private readonly IAcademiesDbContext _academiesDbContext;
+    private readonly ILogger<DataSourceProvider> _logger;
+
+    [ExcludeFromCodeCoverage]
+    public DataSourceProvider(AcademiesDbContext academiesDbContext, ILogger<DataSourceProvider> logger) : this(
+        (IAcademiesDbContext)academiesDbContext, logger)
+    {
+    }
+
+    public DataSourceProvider(IAcademiesDbContext academiesDbContext, ILogger<DataSourceProvider> logger)
+    {
+        _academiesDbContext = academiesDbContext;
+        _logger = logger;
+    }
+
+    public async Task<DataSource> GetGiasUpdated()
```

Comment:
This method (and `GetMstrUpdated` and `GetCdmUpdated`) are very similar - can we refactor the common code into a private method that they all use?